test/src/drivers/test_rsvd.cc

```diff
@@ -0,0 +1,160 @@
+
+    virtual void SetUp() {};
+
+    virtual void TearDown() {};
+
+    template <typename T>
+    static void test_RSVD1_general(int64_t m, int64_t n, int64_t k, int64_t p, int64_t block_sz, T tol, std::tuple<int, T, bool> mat_type, uint32_t seed) {
+        
+        printf("|==================================TEST QB2 GENERAL BEGIN==================================|\n");
+        using namespace blas;
+        using namespace lapack;
+        
+        // For running QB
+        std::vector<T> A(m * n, 0.0);
+        gen_mat_type<T>(m, n, A, k, seed, mat_type);
+
+        int64_t size = m * n;
+
+        // For results comparison
+        std::vector<T> A_hat(size, 0.0);
+        std::vector<T> A_hat_buf (m * k, 0.0);
+        std::vector<T> A_k(size, 0.0);
+        std::vector<T> A_cpy (m * n, 0.0);
+
+        // For RSVD
+        std::vector<T> s1(n, 0.0);
+        std::vector<T> S1(n * n, 0.0);
+        std::vector<T> U1(m * n, 0.0);
+        std::vector<T> VT1(n * n, 0.0);
+
+        // For low-rank SVD
+        std::vector<T> s(n, 0.0);
+        std::vector<T> S(n * n, 0.0);
+        std::vector<T> U(m * n, 0.0);
+        std::vector<T> VT(n * n, 0.0);
+
+        T* A_dat = A.data();
+
+        T* A_hat_dat = A_hat.data();
+        T* A_hat_buf_dat = A_hat_buf.data();
+        T* A_k_dat = A_k.data();
+        T* A_cpy_dat = A_cpy.data();
+
+        T* U1_dat = U1.data();
+        T* s1_dat = s1.data();
+        T* S1_dat = S1.data();
+        T* VT1_dat = VT1.data();
+
+        T* U_dat = U.data();
+        T* s_dat = s.data();
+        T* S_dat = S.data();
+        T* VT_dat = VT.data();
+
+        // Create a copy of the original matrix
+        copy(size, A_dat, 1, A_cpy_dat, 1);
+
+        //Subroutine parameters 
+        bool verbosity = false;
+        bool cond_check = true;
+        bool orth_check = true;
+        int64_t passes_per_iteration = 1;
+
+        // Make subroutine objects
+        // Stabilization Constructor - Choose PLU
+        Stab<T> Stab(use_PLUL, cond_check, verbosity);
+
+        // RowSketcher constructor - Choose default (rs1)
+        RS<T> RS(Stab, seed, p, passes_per_iteration, verbosity, cond_check, use_rs1);
+
+        // Orthogonalization Constructor - Choose CholQR
+        Orth<T> Orth_RF(use_CholQRQ, cond_check, verbosity);
+
+        // RangeFinder constructor - Choose default (rf1)
+        RF<T> RF(RS, Orth_RF, verbosity, cond_check, use_rf1);
+
+        // Orthogonalization Constructor - Choose CholQR
+        Orth<T> Orth_QB(use_CholQRQ, cond_check, verbosity);
+
+        // QB constructor - Choose defaut (QB2)
+        QB<T> QB(RF, Orth_QB, verbosity, orth_check, use_qb2);
+
+        // RSVD constructor - Choose defaut (RSVD1)
+        RSVD<T> RSVD(QB, verbosity, use_rsvd1);
+
+        // Regular QB2 call
+        RSVD.call(
+            m,
+            n,
+            A,
+            k,
+            block_sz,
+            tol,
+            U1,
+            s1,
+            VT1
+        );
```

Comment:
Okay, this looks excessive.

Reply:
Meaning this should go on one line?

Reply:
This could probably fit cleanly on one line. But if you wanted to split it up over two lines that would be okay.